pkg/controllers/logs/controller_test.go

```diff
@@ -0,0 +1,245 @@
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/ViaQ/log-exploration-api/pkg/elastic"
+	"github.com/gin-gonic/gin"
+	"go.uber.org/zap"
+)
+
+func Test_ControllerFilterByIndex(t *testing.T) {
+	tests := []struct {
+		TestName   string
+		Index      string
+		ShouldFail bool
+		TestData   []string
+	}{
+		{
+			"Filter by App index",
+			"app",
+			false,
+			[]string{"test-log-1", "test-log-2", "test-log-3"},
+		},
+	}
+
+	provider := elastic.NewMockedElastisearchProvider()
+	gin.SetMode(gin.TestMode)
+	router := gin.Default()
+	NewLogsController(zap.L(), provider, router)
+
+	for _, tt := range tests {
+		t.Log("Running:", tt.TestName)
+
+		provider.PutDataIntoIndex(tt.Index, tt.TestData)
+
+		rr := httptest.NewRecorder()
+		req, err := http.NewRequest(http.MethodGet, "/logs/indexfilter/app", nil)
+		if err != nil {
+			t.Errorf("Failed to create HTTP request. E: %v", err)
+		}
+		router.ServeHTTP(rr, req)
+		resp := rr.Body.String()
+
+		expected, err := json.Marshal(map[string]interface{}{"Logs": tt.TestData})
+		if err != nil {
+			t.Errorf("failed to marshal test data. E: %v", err)
+		}
+		expectedResp := string(expected)
+
+		if resp != expectedResp {
+			t.Errorf("expected response to be %s, got %s", expectedResp, resp)
+		}
+	}
+
+}
+
+func Test_ControllerGetAllLogs(t *testing.T) {
+	tests := []struct {
+		TestName   string
+		Index      string
+		ShouldFail bool
+		TestData   []string
+	}{
+		{
+			"Get all logs",
+			"infra",
+			false,
+			[]string{"test-log-1", "test-log-2", "test-log-3"},
+		},
+	}
+
+	provider := elastic.NewMockedElastisearchProvider()
+	gin.SetMode(gin.TestMode)
+	router := gin.Default()
+	NewLogsController(zap.L(), provider, router)
+
+	for _, tt := range tests {
+		t.Log("Running:", tt.TestName)
+
+		provider.PutDataIntoIndex(tt.Index, tt.TestData)
+
+		rr := httptest.NewRecorder()
+		req, err := http.NewRequest(http.MethodGet, "/logs/", nil)
+		if err != nil {
+			t.Errorf("Failed to create HTTP request. E: %v", err)
+		}
+		router.ServeHTTP(rr, req)
+		resp := rr.Body.String()
+
+		expected, err := json.Marshal(map[string]interface{}{"Logs": tt.TestData})
+		if err != nil {
+			t.Errorf("failed to marshal test data. E: %v", err)
+		}
+		expectedResp := string(expected)
+
+		if resp != expectedResp {
+			t.Errorf("expected response to be %s, got %s", expectedResp, resp)
+		}
+	}
+
+}
+
+
+func Test_ControllerFilterByTime(t *testing.T) {
+	tests := []struct {
+		TestName   string
+		Index      string
+		ShouldFail bool
+		TestData   []string
+	}{
+		{
+			"Filter by time",
+			"infra",
+			false,
+			[]string{"test-log-1", "test-log-2", "test-log-3"},
+		},
+	}
+
+	provider := elastic.NewMockedElastisearchProvider()
+	gin.SetMode(gin.TestMode)
+	router := gin.Default()
+	NewLogsController(zap.L(), provider, router)
+
+	for _, tt := range tests {
+		t.Log("Running:", tt.TestName)
+
+		logTime,_ := time.Parse(time.RFC3339Nano, "2021-03-17T14:22:40+05:30")
+		provider.PutDataAtTime(logTime, tt.TestData)
```

Comment:
Does this method always put data into the `infra` index? Maybe adding an `index` parameter will make it more flexible? So that you can do this with other indices.